File: 프로그래머스/unrated/138476. 귤 고르기/귤 고르기.java
import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        HashMap<Integer, Integer> hm = new HashMap<>();
        
        for(int i = 0; i < tangerine.length; i++){
            hm.put(tangerine[i], hm.getOrDefault(tangerine[i],0)+1);
        }
        
        List<Integer> valueList = new ArrayList<>(hm.values());
        valueList.sort(new Comparator<Integer>(){
            @Override
            public int compare(Integer o1, Integer o2){
                return o2.compareTo(o1);
            }
        });
        
        while(k>0){
            k = k - valueList.get(0);
            valueList.remove(0);
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

#define MAX 10000001

int cnt[MAX];
int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
    for (int size : tangerine) {
        cnt[size - 1]++;
    }
    
    sort(cnt, cnt + MAX, greater<int>());
    
    for (int i = 0; i < MAX; i++) {
        int c = cnt[i];
        k -= c;
        answer++;
        
        if (k <= 0) return answer;
    }
    return answer;
}